keps/sig-instrumentation/draft-20190404-kubernetes-control-plane-metrics-stability.md

```diff
@@ -0,0 +1,250 @@
+---
+title: Kubernetes Control-Plane Metrics Stability
+authors:
+  - "@logicalhan"
+owning-sig:
+  - sig-instrumentation
+participating-sigs:
+  - sig-instrumentation
+  - sig-api-machinery
+  - sig-node
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2019-04-04
+see-also:
+  - 0031-kubernetes-metrics-overhaul
+status: provisional
+---
+
+# Kubernetes Control-Plane Metrics Stability
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Background](#background)
+* [Proposal](#proposal)
+    * [Metric Definition Phase](#metric-definition-phase)
+    * [Metric Instantiation Phase](#metric-instantiation-phase)
+    * [Metric Enrollment Phase](#metric-enrollment-phase)
+* [Design Details](#design-details)
+  * [Test Plan](#test-plan)
+  * [Graduation Criteria](#graduation-criteria)
+* [Drawbacks](#drawbacks)
+* [Alternative](#alternatives)
+* [Unresolved Questions](#unresolved-questions)
+* [Implementation History](#implementation-history)
+* [Reference](#reference)
+
+## Summary
+
+Currently metrics emitted in the kubernetes control-plane do not offer any stability guarantees. This Kubernetes Enhancement Proposal (KEP) proposes a strategy and framework for programmatically expressing how stable a metric is. This KEP also defines the specific guarantees made for each enumerated level of stability. Since this document will likely evolve with ongoing discussion around metric stability, it will be updated accordingly.
```

Comment:
Instead of "stability", maybe say "lifecycle" or "support" or "future compatibility".  Stability in this particular context makes me think about the stability of the values (like do they measure a stable quantity), which isn't what we mean.

Reply:
I don't particularly mind renaming it but I am concerned it may end up being more confusing. The current phrasing is historically consistent with the way the issue has been described. 

Within sig-instrumentation, this issue has historically been referenced as an issue of metric stability. For instance, the [metrics overhaul kep](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/0031-kubernetes-metrics-overhaul.md#motivation) refers to the effort as "put[ting] us a step closer to having stability guarantees for Kubernetes around metrics" and that currently metrics are "excluded from any kind of stability requirements".

Reply:
If the intended audience isn't confused, this SGTM.

Reply:
Sure thing.